jitsi-feedback-plugin/feedback.js

```diff
@@ -132,8 +131,121 @@
                     window.APP.conference.feedbackToken = feedbackToken;
                 })
                 .catch(e => console.error(`${LOG} failed to fetch JWT`, e));
+                
             return;
         }
+
+        _gatherMetrics() {
+            const metrics = {};
+            const config = APP.store.getState()['features/base/config'];
+            const flags = config.metadata || [];
+            
+            try {
+                flags.forEach(flag => this._addMetric(flag, metrics));
+            } catch (e) {
+                console.error(`${LOG} Metrics error:`, e);
+                return {};
+            }
+            
+            return metrics;
+        }
+
+        _getMatrixContext() {
+            const token = window.APP.store.getState()['features/base/jwt'].jwt;
+            const payload = token.split('.')[1];
+            const content = JSON.parse(atob(payload));
+            return content.context;
+        }
+
+        _addMetric(flag, metrics)  {
+            const config = APP.store.getState()['features/base/config'];
+            const conference = window.APP.store.getState()['features/base/conference'].conference;
+            const localParticipant = window.APP.store.getState()['features/base/participants'].local;
+
+            switch (flag) {
+                // meetingId
+                case 'MEETING_URL':
+                    metrics.meetingUrl = window.location.href;
+                    break;
+                case 'MEETING_ID':
+                    metrics.meetingId  = JitsiMeetJS.analytics.permanentProperties.conference_name;
+                    break;
+
+                // participantID    
+                case 'PARTICIPANT_ID':
+                    metrics.participantId = conference.myUserId();
+                    break;
+
+                // matrix user Id    
+                case 'MATRIX_USER_ID':
+                    metrics.matrixUserId = localParticipant.email;
+                    break;
+                case 'DISPLAY_NAME':
+                    metrics.displayName  = localParticipant.name;
+                    break;
+
+                case 'USER_REGION':
+                    metrics.userRegion = JitsiMeetJS.analytics.permanentProperties.userRegion;
+                    break;
+                
+                // app data
+                case 'APP_LIB_VERSION':
+                    metrics.appLibVersion = JitsiMeetJS.version;
+                    break;   
+                case 'APP_FOCUS_VERSION':
+                    metrics.appFocusVersion = conference.componentsVersions.versions.focus;
```

Comment:
`FEEDBACK_HANDLER Metrics error: TypeError: Cannot read properties of undefined (reading 'componentsVersions')` when submitting feedback using the popup that appears after hanging up. In these cases no metadata is received by the database at all.

Likely the try-catch block needs to be moved into the forEach (line 144).